packages/apollo-server-core/src/ApolloServer.ts

```diff
@@ -315,58 +349,57 @@ export class ApolloServerBase {
 
     this.parseOptions = parseOptions;
 
-    if (mocks || (typeof mockEntireSchema !== 'undefined' && mocks !== false)) {
-      addMockFunctionsToSchema({
-        schema: this.schema,
-        mocks:
-          typeof mocks === 'boolean' || typeof mocks === 'undefined'
-            ? {}
-            : mocks,
-        preserveResolvers:
-          typeof mockEntireSchema === 'undefined' ? false : !mockEntireSchema,
-      });
-    }
-
-    // The schema hash is a string representation of the shape of the schema
-    // it is used for reporting and can be used for a cache key if needed
-    this.schemaHash = generateSchemaHash(this.schema);
-
     // Note: doRunQuery will add its own extensions if you set tracing,
     // or cacheControl.
     this.extensions = [];
 
-    // In an effort to avoid over-exposing the API key itself, extract the
-    // service ID from the API key for plugins which only needs service ID.
-    // The truthyness of this value can also be used in other forks of logic
-    // related to Engine, as is the case with EngineReportingAgent just below.
-    this.engineServiceId = getEngineServiceId(engine);
+    const updateSchemaDerivedData = async () => {
```

Comment:
In a pursuit of reducing the overall height of this `constructor`, is making this a `private` method a possibility?  I believe that would require making the `config` available as a property of the `ApolloServerBase` class, but that seems valuable in its own way.

I don't think we should do it within this PR, but a follow-up commit seems reasonable.

Reply:
Address in follow up, possibly with the move to the promise returning all the derived data?